File: maths/SumOfDigits.py
# the below function returns sum of digits of a number
def SumOfDigits(n) -> int:
    sum_of_digits = 0

    while n!=0:
        sum_of_digits += (n%10)
        # remove floating values after division
        n = int(n/10)

    return sum_of_digits

print(SumOfDigits(1234)) # O/P : 10

File: maths/SumOfDigits.cpp
#include <bits/stdc++.h>

using namespace std;

// function returns sum of digits
int sumOfDigits(int n){
    int sum = 0;

    // looping through each digit
    while (n != 0){
        sum += (n%10); // 1 + 2 + 3
        n /= 10;
    }

    return sum // 1+2+3 = 6
}
int main(){
    cout<<sumOfDigits(123);// O/P: 6
    return 0;
}